src/main/java/org/fogbowcloud/manager/core/rest/services/ComputeOrdersService.java

```diff
@@ -0,0 +1,75 @@
+package org.fogbowcloud.manager.core.rest.services;
+
+import org.fogbowcloud.manager.core.controllers.ApplicationController;
+import org.fogbowcloud.manager.core.datastructures.SharedOrderHolders;
+import org.fogbowcloud.manager.core.exceptions.ComputeOrdersServiceException;
+import org.fogbowcloud.manager.core.models.orders.ComputeOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Service
+public class ComputeOrdersService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ComputeOrdersService.class);
+
+    private ApplicationController applicationController;
+    private SharedOrderHolders ordersHolder;
+
+    public ComputeOrdersService() {
+        applicationController = ApplicationController.getInstance();
+        ordersHolder = SharedOrderHolders.getInstance();
+    }
+
+    public ResponseEntity<Order> createCompute(ComputeOrder computeOrder, String accessId, String localTokenId) {
+
+        if (computeOrder == null) {
+            return new ResponseEntity<Order>(HttpStatus.BAD_REQUEST);
+        }
+        try {
+            Token federatedToken = applicationController.authenticate(accessId);
+            Token localToken = createLocalToken(localTokenId);
+            setOrderTokens(computeOrder, localToken, federatedToken);
+            computeOrder.setOrderState(OrderState.OPEN);
```

Comment:
Why you are injecting this dependency here? I think that the Orders constructor should be responsible for it.

Reply:
Terrible mistake here. Sorry.